src/main/kotlin/org/rust/cargo/runconfig/command/CargoCommandConfiguration.kt

```diff
@@ -212,6 +218,52 @@ class CargoCommandConfiguration(
         fun findCargoProject(project: Project, cmd: String, workingDirectory: Path?): CargoProject? = findCargoProject(
             project, ParametersListUtil.parse(cmd), workingDirectory
         )
+
+        fun findCargoPackage(
+            cargoProject: CargoProject,
+            additionalArgs: List<String>,
+            workingDirectory: Path?
+        ): CargoWorkspace.Package? {
+            val packages = cargoProject.workspace?.packages.orEmpty()
```

Comment:
Do I understand correctly that it's impossible to run Cargo with a package outside of a workspace? Hence, we could filter workspace-only packages here.